Fill.py

```diff
@@ -129,18 +142,35 @@ def distribute_items_restrictive(world: MultiWorld, fill_locations=None):
         else:
             restitempool.append(item)
 
-    world.random.shuffle(fill_locations)
-    call_all(world, "fill_hook", progitempool, nonexcludeditempool, localrestitempool, nonlocalrestitempool, restitempool, fill_locations)
+    call_all(world, "fill_hook", progitempool, nonexcludeditempool,
```

Comment:
Thhe fill hook now gets called before the locations are shuffled. Any hooks that relied on this behaviour would now be broken.

Reply:
I moved the shuffle to line 111 so it would happen before I categorized locations, so they should still be shuffled in order. However, we don't go through the locations in the same order that they appear in fill_locations. Are there worlds that rely on the order of fill_locations for their logic?  Should I append the three arrays and send it in here instead to try to get closer to the actual processing order?

Reply:
I see now. That's a hard question to answer without some test runs to see it still produces properly random distribution.

Reply:
I believe since we are both shuffling the locations and items, we have two layers that should help us ensure that we get a pretty good random distribution.  I also added a couple of tests that show multiple world item distributions with and without priority locations. I'm not sure how else to measure the distribution, since it is by nature random.